packages/react-divider/src/components/Divider/useDivider.ts

```diff
@@ -23,6 +23,8 @@ export const useDivider = (
   const state = mergeProps(
     {
       ref: useMergedRefs(ref, React.useRef(null)),
+      /* The Id created to expose accessability for readers */
+      labelledById: props.children ? useId('divider-') : undefined,
```

Comment:
⚠️ This violates rules of hooks, they cannot be conditional: https://reactjs.org/docs/hooks-rules.html

Reply:
A proper solution will be move out `useId()`:


```diff
+  const dividerId = useId('divider-')
  const state = mergeProps(
    {
      ref: useMergedRefs(ref, React.useRef(null)),
      /* The Id created to expose accessability for readers */
-      labelledById: props.children ? useId('divider-') : undefined,
+      labelledById: props.children ? dividerId : undefined, 
```

Reply:
Updated